### ISSUE TITLE
Many features needed :-

### ISSUE THREAD

**Issue by Zaxx69** [AmrDeveloper/LinkHub#50]
1. Amoled black theme 

2. Folder in folder feature 

3. Catogories :-
   . Most used 
   . Last added 
   . Favourites 
   . Uncategorized 

4. Copy link by swiping to the right
 
5. Some ui changes and optimization. You can find some inspiration from keeplink app https://play.google.com/store/apps/details?id=org.erikjaen.tidylinksv2

6. Search by tag ( we can name tags and assign them to links) 

The app is otherwise great, thanks for making this app, please consider about the above recommendations. I think that it would make the app for powerful and useful

**Comment by AmrDeveloper** [comment]
Thank you @Zaxx69, I will check the recommendation

**Comment by Zaxx69** [comment]
Any updates on it?

**Comment by Zaxx69** [comment]
@AmrDeveloper any updates on it?

**Comment by AmrDeveloper** [comment]
Hello @Zaxx69,

Sorry for delay and ping me anytime, I added them to todo list but we need to think and manage the development correctly for example supporting nested folders or add tags will need more testing for export and import features.

If you have any suggestion or design for the Amoled black theme please share it, also if we will support other theme with dark and white we need to be ready for more themes in future

Thanks
Amr Hesham

**Comment by Zaxx69** [comment]
@AmrDeveloper  I can help with testing with nested folder and it's ui and probably give some ideas and for the black theme it's pretty simple
1. Black oled background
2. A primary color (users choice)
3. Material you theme and switches 

Example of black theme :
![Screenshot_2023-10-14-21-01-34-07_a13e17678c56528bd2f18fcaaa8af4ba.jpg](https://github.com/AmrDeveloper/LinkHub/assets/138523882/61d36f51-cc1f-44a1-ba79-bc1bf040b981)

Example of material you switches :
![Screenshot_2023-10-14-21-01-14-76_a13e17678c56528bd2f18fcaaa8af4ba.jpg](https://github.com/AmrDeveloper/LinkHub/assets/138523882/965753c6-9bd6-46ca-991c-a511d017c3ce)

For the nested folder part ,here are my suggestions
1. Round cornered rectangle tiles (2 or 3 tiles in a row) with names on it,for example let's name it youtube shorts .

Example of rounded rectangle tiles 
![Screenshot_2023-10-14-21-02-23-87_dba045b72264cb60f98b4e733fab6094.jpg](https://github.com/AmrDeveloper/LinkHub/assets/138523882/747868e0-60f7-4d39-b26a-dbf1f10962e8)

But make it like this :
![Screenshot_2023-10-14-21-00-47-31_a13e17678c56528bd2f18fcaaa8af4ba.jpg](https://github.com/AmrDeveloper/LinkHub/assets/138523882/b77e306f-72f7-4d7b-9e98-6b5b8c778fff)


2. After opening a folder ,there should be two choices
  .either you can paste the link directly
  .or you can create a subfolder within it
The process repeats

Example of featured options 
![Screenshot_2023-10-14-21-02-17-97_dba045b72264cb60f98b4e733fab6094.jpg](https://github.com/AmrDeveloper/LinkHub/assets/138523882/f5e2bf12-55e6-4da1-95f6-b5a929c10278)



3. There should a edit icon for Both the folder and links .
For example
I created a yt shorts folder ,now using the edit button ,i can change the name and colour of the tiles ,create a sub folder ,attach a tag

Whereas
Using the edit button of the link ,i can edit the url of the link ,attach a tag ,transfer / migrate it to another folder / subfolder ,copy / share it ,give it a name

4. A search in main page to search  folders ,subfolders ,links by using tag ,name ,component of the link 

5. A search button in sub folders to find the same except folders present in main page

6. Swipe the link towards right to copy the link and delete it by swiping towards left and a undo button should pop up to revert it incase of a mistake

I will help with all my might ,it would be better if we can talk in either telegram and discord because i can't explain everything in here

**Comment by AmrDeveloper** [comment]
- I have implemented dynamic theme before in other project it's good but we need to think to support it or improving current themes

- For the sub folder feature i think we need a better design that the current floating button.

- Add edit button for each item make the UI not simple, you can just long click and edit them.

- I agree with you that this type of search will be very good too but need some restructure.

- The problem with add different action for each direction is that you need to remember them correctly and i want the user to don't think or remember and every think should be very simple

Thanks a lot for your helpful suggestions @Zaxx69, You can find me to telegram or discord with username amrdeveloper similar to Github feel free to ping me and we can have long discussion

**Comment by Thewisem** [comment]
So Any update?

**Comment by AdityaMH** [comment]
Most important feature is protected folder. So you need input password to open a folder.

**Comment by AmrDeveloper** [comment]
> Most important feature is protected folder. So you need input password to open a folder.
> 
> > So Any update?
> 
> No

We can move it to new issue so if someone want to contribute, currently i am trying to solve update issue on google play so if any one interested you can implement it

**Comment by AmrDeveloper** [comment]
All the requested features are good, also the new ones like sorting folders, nested folders, i think before we can start working on them, a small refactor maybe helpful to make it easy to support more features and themes, maybe more to Jetpack compose too

**Comment by OctoberNicole** [comment]
I think it would be really nice to be able to directly save and manage small bits of data, like stickers or code snippets, cropped screenshots & small images like icons. I think I'm dreaming though.
